force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -163,9 +186,10 @@ public with sharing class ClassesObjectsMethods {
      */
     public static Integer generateRandomNumber() {
         // Use Math class to generate number between 50 and 100
-
+        Double random = math.random();
+        Integer randomnumber = 50 + (Integer)(random*50);
```

Comment:
Be sure to use camelcase for variables with more than one word: `randomNumber`.

Also, following up on the last comment, your final formula for a range of 50 to 100 should look like this:
`Integer randomNumber = (Integer) (Math.random() * 51) + 50;`

Your current formula would never generate 100 as a potential result.